Add null / undefined check to core::joiner

NandoTheessen opened the pull request:
References #80 

This PR adds a null / undefined check to the core::joiner function & returns null in case those are encountered.
It prevents react-csv from crashing the application when presented with data that contains empty indices (which you'd see for example when working with firebase).

It also adds two tests for said function to ensure it's functionality. 

Are there any more tests I should add? @mccabemj

mccabemj commented on src/core.js:
```diff
@@ -50,7 +50,10 @@ export const getHeaderValue = (property, obj) => {
 export const joiner = ((data,separator = ',') =>
- data.map((row, index) => row.map((element) => "\"" + elementOrEmpty(element) + "\"").join(separator)).join(`\n`)
+ data.map((row, index) => {
+  if(row == null) return null
```
Would this not be neater if you used `filter` on the data? instead of returning null in a map @NandoTheessen

mriccid replied:
@mccabemj the difference between the two is that filtering the data removes the empty line from the output, whereas returning `null` in the map function leaves the line in output, albeit as an empty line. 

I am leaning towards filtering null and undefined rows before mapping as you suggested so that they don't appear at all, but I am keen to hear what other users of this library would want

mccabemj replied:
Yeah @mriccid in the issue @NandoTheessen raised, I was leaning towards leaving the blank lines in. I think it would be neater to omit them.

@NandoTheessen you may be best placed to answer this, as you have source data with null lines. What would you expect?

It would be quite interesting to allow blank lines. I can see situations where people would want reports that have blank lines to separate groupings of data. Perhaps we could even add this to the documentation as a feature.

NandoTheessen replied:
Honestly, having no blank lines would be the most intuitive to me. 
The way I've encountered this is when working with firebase (where deleted entries would return null / undefined in the array).

The user doesn't even know that that is going on behind the scenes. For them expected behaviour would indeed be no empty lines.
I'll change my PR to reflect that.

NandoTheessen commented:
@mriccid @mccabemj blank lines are now omitted 👍

mccabemj commented:
@NandoTheessen Sorry to be a pain here, but your PR contains a load of core.js formatting changes unrelated to your actual fix. Your [original commit](https://github.com/react-csv/react-csv/pull/116/commits/b7673ac2f64e41ebac13911c14f60ab154778276) was fine.

If you can remove all formatting to unrelated code, and just include the changes you made to fix the issue. Leave the CND files for me to do.

I understand the issue around formatting, and I have opened a new issue for discussion #118

NandoTheessen commented:
Hey @mccabemj sorry for the formatting changes, prettier went rogue on me there. 
Took me a bit longer then expected, had a busy work week. 
Reverted 2 commits and re-added the solution to the issue.
Also changed the test accordingly.

Let me know if you require any more changes!
